tensorflow/core/kernels/gather_nd_op.cc

```diff
@@ -228,6 +228,8 @@ namespace functor {
   DECLARE_GPU_SPECS_INDEX(T, int32); \
   DECLARE_GPU_SPECS_INDEX(T, int64)
 
+TF_CALL_int32(DECLARE_GPU_SPECS);
```

Comment:
this seems out of scope.  i'm ok with the int64 version, but we tend to not provide int32 variants on gpu.  for example, see gather() which doesn't allow gathers into int32 tensors on gpu.

Reply:
@Androbin maybe int32 support still needs to be reverted?

Reply:
I honestly don't know. It seems to be a design question. [Not long ago](https://github.com/tensorflow/tensorflow/pull/13382#discussion_r143081981), he said:
> I'm perfectly OK with scatter_nd and gather_nd getting GPU kernels for int32/int64, but not scatter/gather.

Reply:
Maybe the team changed opinions, or it's just a misunderstanding here.